sources/api/pluto/src/main.rs

```diff
@@ -144,42 +162,37 @@ async fn get_max_pods(client: &mut ImdsClient) -> Result<String> {
     error::NoInstanceTypeMaxPods { instance_type }.fail()
 }
 
-/// Returns the cluster's DNS IPV4 address. First it attempts to call EKS describe-cluster to find
-/// the `serviceIPv4CIDR`. If that works, it returns the expected cluster DNS IP address which is
-/// obtained by substituting `10` for the last octet. If the EKS call is not successful, it falls
-/// back to using IMDS MAC CIDR blocks to return one of two default addresses.
-async fn get_cluster_dns_ip(client: &mut ImdsClient) -> Result<String> {
-    // try calling eks describe-cluster to figure out the dns cluster ip
-    if let Some(dns_ip) = get_dns_from_eks().await {
-        // we were able to calculate the dns ip from the cidr range we received from eks
-        return Ok(dns_ip);
+/// Returns the cluster's DNS address.
+///
+/// For IPv4 clusters, first it attempts to call EKS describe-cluster to find the `serviceIpv4Cidr`.
+/// If that works, it returns the expected cluster DNS IP address which is obtained by substituting
+/// `10` for the last octet. If the EKS call is not successful, it falls back to using IMDS MAC CIDR
+/// blocks to return one of two default addresses.
+///
+/// For IPv6 clusters, EKS currently is choosing a fixed unique local address (ULA) of `fdf9:2874:204b::/108`
+/// for cluster IPs. Cluster DNS IP in this case would always be hardcoded to `fdf9:2874:204b::a`.
+/// (The `::a` is equivalent to `.10` for IPv4)
+/// FIXME Update this once EKS changes their approach for determining the IPv6 service CIDR
+async fn get_cluster_dns_ip(
+    client: &mut ImdsClient,
+    network_config: KubernetesNetworkConfigResponse,
+) -> Result<String> {
+    if network_config.ip_family == Some("ipv6".to_string()) {
+        // For IPv6 clusters, `::a` is equivalent to `.10` for IPv4
+        return Ok(
+            DEFAULT_SERVICE_IPV6_CIDR.split('/').collect::<Vec<&str>>()[0].to_string() + "a",
```

Comment:
I wanted to show how the cluster DNS IP is derived from the service IPv6 CIDR. (i.e. setting the last hextet to `a`). This value is the way it is because EKS has chosen a fixed ULA for the default service IPv6 CIDR.

Reply:
Also, we'd need this manipulation down the road once EKS decides to not hardcode the service IPv6 CIDR and just generate a unique one per cluster (which is likely).

Reply:
EKS changed approach to generating a random ULA for each cluster instead of hardcoding it. So now I'm using `serviceIpv6Cidr` directly